### PR TITLE
Persistence filestorage integration

### PR DESCRIPTION
**Pull request by egrozdani**

### CODE REVIEW

File: aks/filestorage/templates/file-storage-api-cloud-cm.yaml [hunk 1]
```diff
@@ -26,4 +26,5 @@ data:
           account.name: vtchk8sblueprinttfst
           container.name: filestorage
           account.key: ${STORAGE_ACCOUNT_KEY}
-          connection: ${STORAGE_CONNECTION}
\ No newline at end of file
+          connection: ${STORAGE_CONNECTION}
+      persistence.url: http://persistence-service.default.svc.cluster.local:8081/api/v1/messages
```
Comment (lrohner) [inline aks/filestorage/templates/file-storage-api-cloud-cm.yaml:30]: wrong indentation

File: file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/config/RestTemplateConfig.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+
+@Configuration
+public class RestTemplateConfig {
```
Comment (lrohner) [inline file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/config/RestTemplateConfig.java:12]: Isn't the rest template by default available?

File: file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/service/AuditingService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    HttpHeaders headers = populateHeaders();
+    HttpEntity<AuditingRequestDto> request = new HttpEntity<>(auditingRequest, headers);
+
+    postForEntity(request);
+  }
+
+  private HttpHeaders populateHeaders() {
+    HttpHeaders headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    headers.setBearerAuth(getToken().get());
```
Comment (lrohner) [inline file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/service/AuditingService.java:46]: possible nullpointer as directly using .get()
only set the header if the token is present

File: intellij-java-google-style.xml [hunk 4]
```diff
@@ -0,0 +1,598 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lrohner) [inline intellij-java-google-style.xml:1]: Is this similar to https://git.valtech.swiss/projects/IC/repos/code-conventions/browse/java ? Or why adding it?

File: persistence/src/main/java/ch/valtech/kubernetes/microservice/cluster/persistence/service/PersistenceService.java [hunk 5]
```diff
@@ -1,59 +1,38 @@
...
-  private Auditing addAuditRecord(AuditingRequestDto requestDto) {
-    Auditing auditing = new Auditing();
-    auditing.setUser(getOrCreateUser(requestDto.getEmail()));
-    auditing.setMessage(new Message(requestDto.getKey(), requestDto.getMessageValue()));
-    auditing.setModificationDate(LocalDate.now());
-    auditingRepository.save(auditing);
-    return auditing;
+  private Auditing addAuditRecord(AuditingRequestDto requestDTO) {
+    return auditingRepository.save(
+        persistenceMapper.toAuditing(requestDTO, getUsername().get()));
```
Comment (lrohner) [inline persistence/src/main/java/ch/valtech/kubernetes/microservice/cluster/persistence/service/PersistenceService.java:36]: getUsername could be empty

**Review by lrohner — CHANGES_REQUESTED** [review]

File: file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/config/RestTemplateConfig.java (hunk 2 above)
Comment (egrozdani) [reply file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/config/RestTemplateConfig.java:12]: No, you need to define the bean

File: aks/filestorage/templates/file-storage-api-cloud-cm.yaml (hunk 1 above)
Comment (egrozdani) [reply aks/filestorage/templates/file-storage-api-cloud-cm.yaml:30]: Indentation is good. Will move it up for better readability

File: file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/utils/FileStorageUtils.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package ch.valtech.kubernetes.microservice.cluster.filestorage.utils;
+
+import java.util.Optional;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationDetails;
+
+public class FileStorageUtils {
```
Comment (lrohner) [inline file-storage/src/main/java/ch/valtech/kubernetes/microservice/cluster/filestorage/utils/FileStorageUtils.java:8]: public final and private constructor missing.
You can use @NoArgsConstructor(access = AccessLevel.PRIVATE) for constructor

File: persistence/src/main/java/ch/valtech/kubernetes/microservice/cluster/persistence/service/PersistenceUtils.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
+package ch.valtech.kubernetes.microservice.cluster.persistence.service;
+
+import ch.valtech.kubernetes.microservice.cluster.persistence.domain.Auditing;
+import java.util.Optional;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+
+public class PersistenceUtils {
```
Comment (lrohner) [inline persistence/src/main/java/ch/valtech/kubernetes/microservice/cluster/persistence/service/PersistenceUtils.java:9]: public final and private constructor missing @NoArgsConstructor(access = AccessLevel.PRIVATE)

**Review by lrohner — CHANGES_REQUESTED** [review]

**Pull request merged**